### ISSUE TITLE
creation of custom system doesn't populate connection secret

### ISSUE THREAD

**Issue by janwillies** [crossplane-contrib/provider-styra#6]
### What happened?
I tried to create a `system` with type `custom`. The system was created correctly at the styra das api, but the connection secret in kubernetes was empty.

I expected to find the api token in there, or the opa conf, so I can register my opa deployment with the styra das system

### How can we reproduce it?
create a custom system like this:
```yaml
apiVersion: styra.crossplane.io/v1alpha1
kind: System
metadata:
  name: custom-system-demo
spec:
  forProvider:
    description: testing custom opa system
    type: custom
  providerConfigRef:
    name: styra-provider
  writeConnectionSecretToRef:
    name: custom-system-demo
    namespace: crossplane-system
```

### What environment did it happen in?
provider-styra version: 7f325a5

**Comment by MisterMX** [comment]
This issue should be fixed by now. Are you running the latest version?

**Comment by janwillies** [comment]
I was running https://github.com/crossplane-contrib/provider-styra/commit/7f325a545c739ba82c5627a1fc399d9bd0e9b851 when I tested.

I'm referring to this part of the code: https://github.com/crossplane-contrib/provider-styra/blob/main/pkg/controller/system/controller.go#L357-L363

Note that I'm trying to create a system of type `custom` here.

**Comment by MisterMX** [comment]
You have to create a system of type `kubernetes` or better `kubernetes:v2`. Only for those does Styra provide a helm chart.

**Comment by janwillies** [comment]
But I want to create a system of type `custom` :) 

I opened the issue to see if that's something `provider-styra` is able to support.

**Comment by MisterMX** [comment]
Sorry, I misunderstood your issue with something that sounded similar.

Yes, you are right, we currently only support Styra's `kubernetes` and `kubernetes:v2` for connection secrets. Instead of the helm values we could fetch the OPA config from `/v1/systems/<system-id>/assets/opa-config` and write it out as connection secret.

**Comment by schroeder-paul** [comment]
@janwillies PoC:

```yaml
$ kubectl -n crossplane-system get secrets connection-details -o yaml        
apiVersion: v1
data:
  opaConfig: ZGlzY292ZXJ ... xL2J1bmRsZXMK
kind: Secret
metadata:
  creationTimestamp: "2022-01-21T15:21:05Z"
  name: connection-details
  namespace: crossplane-system
  ownerReferences:
  - apiVersion: styra.crossplane.io/v1alpha1
    controller: true
    kind: System
    name: example-system
    uid: 93a3842a-ca47-4414-a17e-6b1831761433
  resourceVersion: "1202051"
  uid: eedf4b10-5251-4762-820f-505c47c94c7a
type: connection.crossplane.io/v1alpha1
```

**Comment by MisterMX** [comment]
Fixed in #23
